### DISCUSSION TITLE
Smart linemode with more values

### DISCUSSION THREAD

**Discussion by og900aero** [sxyazi/yazi#532] [4. Ideas]
It would be nice if we could set a linemode that could show several things at the same time, and it would be so smart that if the file/directory was created in this year, then show hours:minutes as the time, if this year earlier, then it would show the year. I attached a picture as an example:

![screenshot_1705500194](https://github.com/sxyazi/yazi/assets/49487376/20a48eb7-2f68-4ab0-bcfb-d5bf9e772990)

It can be seen that it shows the size of the files and the date of file/directory creation. Just put a hyphen for the library instead of size for low resources.
I think it's important to be able to see the size and creation time of the elements at the same time.

**Comment by sxyazi** [comment]
You can [create your own linemode](https://yazi-rs.github.io/docs/configuration/yazi#:~:text=and%20sort_reverse%20only-,linemode,-%3A%20Line%20mode) with a few lines of Lua code:

> In addition, you can also specify any 1 to 20 characters, and extend it within a UI plugin. Which means you can implement your own linemode through the plugin by simply overriding the [Folder:linemode method](https://github.com/sxyazi/yazi/blob/main/yazi-plugin/preset/components/folder.lua).

The API of `File` docs: https://yazi-rs.github.io/docs/plugins/common#file

**Comment by sxyazi** [comment]
BTW If you encounter difficulties during creation, let me know, and I will write an example for you :)

**Reply by og900aero** [reply]
Unfortunately, the example I described doesn't want to work, as if it doesn't take the new code.

**Reply by sxyazi** [reply]
Add these lines to your `~/.config/yazi/init.lua`

```lua
local old_linemode = Folder.linemode
function Folder:linemode(area)
  if cx.active.conf.linemode ~= "my-own" then
    return old_linemode(self, area)
  end

  local lines = {}
  local year = os.date("%Y")
  for _, f in ipairs(self:by_kind(self.CURRENT).window) do
    local time = f.cha.modified // 1
    if time and os.date("%Y", time) == year then
      time = os.date("%b %d  %Y", time)
    else
      time = time and os.date("%b %d %H:%M", time) or ""
    end

    local size = f:size()
    lines[#lines + 1] = ui.Line {
      ui.Span(" "),
      ui.Span(size and ya.readable_size(size):gsub(" ", "") or "-"),
      ui.Span(" "),
      ui.Span(time),
      ui.Span(" "),
    }
  end
  return ui.Paragraph(area, lines):align(ui.Paragraph.RIGHT)
end
```

Then set `linemode = "my-own"` within the `[manager]` of your `yazi.toml`, like this:

```toml
[manager]
linemode = "my-own"
# ...
```

Now you should be able to enjoy with it:


![CleanShot 2024-01-22 at 11 53 35](https://github.com/sxyazi/yazi/assets/17523360/c8a1a258-d8fd-47c7-8234-5432de985e76)

**Reply by og900aero** [reply]
Thx, working great!

**Comment by og900aero** [comment]
I create /home/shyciii/.config/yazi/plugins/custom-linemode.yazi/init.lua file, and copied content from https://github.com/sxyazi/yazi/blob/main/yazi-plugin/preset/components/folder.lua file, then I changed the following line:

```
elseif mode == "mtime" then
			local time = f.cha.modified
			spans[#spans + 1] = ui.Span(time and os.date("%y-%m-%d %H:%M", time // 1) or "")
```
To:
```
elseif mode == "mtime" then
			local time = f.cha.modified
			-- local time = f.cha.created
			spans[#spans + 1] = ui.Span(time and os.date("%b %d %H:%M %y", time // 1) or "")

```
When run yazi and change to mtime with m + m key, and show the old method.
